tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,162 @@
+              is still in the running state 
+    """
+    kill_process(duthost, container_name, process)
+
+    logging.info("Checking whether the {} is still running...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_STOP_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, False),
+                  "{} is stopped unexpectedly".format(container_name))
+    logging.info("{} is running".format(container_name))
+
+@pytest.fixture(scope="module", autouse=True)
+def change_autorestart_state(duthost):
+    container_list = get_autorestart_container_list(duthost)
+    for container_name in container_list:
+        logging.info("Change {} auto-restart state to 'enabled'".format(container_name))
+        duthost.shell("config container feature autorestart %s enabled" % container_name)
+
+def test_containers_autorestart(duthost):
+    container_list = get_autorestart_container_list(duthost)
+    for container_name in container_list:
+        logging.info(container_name)
+        if container_name != "restapi":
+            verify_no_autorestart_with_non_critical_process(duthost, container_name, "rsyslogd")
+
+        if container_name == "swss" :
+            verify_autorestart_with_critical_process(duthost, container_name, "portsyncd")
+            time.sleep(7)
+        elif container_name == "lldp":
+            verify_autorestart_with_critical_process(duthost, container_name, "lldpmgrd")
+        elif container_name == "sflow":
+            verify_autorestart_with_critical_process(duthost, container_name, "sflowmgrd")
+        elif container_name == "database":
+            verify_autorestart_with_critical_process(duthost, container_name, "redis")
+        elif container_name == "telemetry":
+            verify_autorestart_with_critical_process(duthost, container_name, "telemetry")
+        elif container_name == "snmp":
+            verify_autorestart_with_critical_process(duthost, container_name, "snmpd")
+        elif container_name == "bgp":
+            verify_autorestart_with_critical_process(duthost, container_name, "zebra")
+        elif container_name == "radv":
+            pass
+            #verify_autorestart_with_critical_process(duthost, container_name, "radvd")
+        elif container_name == "dhcp_relay":
+            verify_autorestart_with_critical_process(duthost, container_name, "dhcrelay")
+        elif container_name == "nat":
+            verify_autorestart_with_critical_process(duthost, container_name, "natmgrd")
+        elif container_name == "teamd":
+            verify_autorestart_with_critical_process(duthost, container_name, "teammgrd")
+        elif container_name == "syncd":
+            verify_autorestart_with_critical_process(duthost, container_name, "syncd")
+        elif container_name == "pmon":
+            pass
+            #verify_autorestart_with_critical_process(duthost, container_name, "ledd")
```

Comment:
This if/elif block requires future maintenance whenever adding/removing conatiners and/or critical processes, defeating the purpose of automating the test using `get_autorestart_container_list`. This should be smarter. For the basic test, it should read the critical_processes file inside the container and choose the first process in the file as the process to kill.

Reply:
Great suggestion! For dhcp_relay, the file critical_processes file includes a single group name isc-dhcp-relay. So I need consider this case.

Reply:
Yes, the dhcp_relay container is a special case, and we may use group names in other containers in the future, so we will need a solution for these cases, also.